### PR TITLE
Add support for vim, emacs, and sublime key maps

### PR DESCRIPTION
**Pull request by HeroicEric**
fixes https://github.com/ember-cli/ember-twiddle/issues/48

vim-mode

~~Right now ember-twiddle is dependent on a fork of ivy-codemirror until https://github.com/IvyApp/ivy-codemirror/pull/5 is merged in.~~

~~This is additionally dependent on https://github.com/IvyApp/ivy-codemirror/pull/7 being merged, which adds support for importing the additional key maps. It is currently working while using my fork that includes the changes in both of those PRs but could also be made to work by manually importing those key map files via bower.~~

It would probably also be cool to persist these settings somewhere but I'm not sure where you would want that done. Maybe in localStorage or a specially named gist?

### CODE REVIEW

File: app/gist/template.hbs [hunk 1]
```diff
@@ -38,6 +38,17 @@
...
+      <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+        Mode <b class="caret"></b>
+      </a>
+      <ul class="dropdown-menu dropdown-menu-right">
+        <li><a {{action 'setEditorKeyMap' 'basic'}}>Basic</a></li>
+        <li><a {{action 'setEditorKeyMap' 'vim'}}>Vim</a></li>
+        <li><a {{action 'setEditorKeyMap' 'emacs'}}>Emacs</a></li>
+        <li><a {{action 'setEditorKeyMap' 'sublime'}}>Sublime</a></li>
+      </ul>
+    </li>
```
Comment (Gaurav0) [inline app/gist/template.hbs]: Would you mind refactoring this into a component?

**Comment by Gaurav0** [comment]
Love this. We just need some tests. Thanks so much.

**Comment by stefanpenner** [comment]
Oh boy!!

**Comment by joostdevries** [comment]
Could you merge master/rebase?

**Comment by HeroicEric** [comment]
@joostdevries it's now up to date with master. Haven't had time to work on persisting to localStorage yet

**Comment by HeroicEric** [comment]
Added some tests and support for persisting your settings to localStorage

**Comment by stefanpenner** [comment]
yay :)

**Comment by Gaurav0** [comment]
@HeroicEric This is awesome. Thanks!

**Comment by stefanpenner** [comment]
the vim mode doesn't appear to have some of the text block support, the example at http://ace.c9.io/build/kitchen-sink.html has @HeroicEric thoughts?

**Comment by HeroicEric** [comment]
@stefanpenner I think that's using a totally different editor [Ace](http://ace.c9.io/) while ember-twiddle is set up with [CodeMirror](https://codemirror.net/).

We could look into switching maybe but we'd probably want to do a feature comparison to see what we'd be trading.

It also looks like there is a [Vim Mode API](https://codemirror.net/doc/manual.html#vimapi) for Codemirror's vim config that we can maybe tweak

**Comment by stefanpenner** [comment]
> ember-twiddle is set up with CodeMirror.

oh for some reason i thought it was ace.

**Comment by stefanpenner** [comment]
can confirm http://codemirror.net/demo/vim.html has the text-block support, i'll see if i can figure out if its an option or something.

**Comment by stefanpenner** [comment]
it appears the only config they are using is:

```
     lineNumbers: true,
        mode: "text/x-csrc",
        keyMap: "vim",
        matchBrackets: true,
        showCursorWhenSelecting: true
```

**Comment by HeroicEric** [comment]
@stefanpenner I can look into it later. It's possible that it's an option not related to vim specifically

**Pull request merged**